packages/react-components/react-tag-picker/library/src/components/TagPickerControl/useTagPickerControl.tsx

```diff
@@ -122,4 +127,74 @@ export const useTagPickerControl_unstable = (
     disabled,
     invalid,
   };
+
+  const { targetDocument } = useFluent();
+  const win = targetDocument?.defaultView;
+
+  React.useEffect(() => {
+    if (!win || !triggerRef.current || !state.expandIcon) {return;}
+
+    // If aria-label or aria-labelledby changes, recalculate aria-label and aria-labelledby for the expandIcon
+    const getExpandLabel = (ariaLabel?: string | null, ariaLabelledBy?: string | null) => {
+      let expandAriaLabel = '';
+      let expandAriaLabelledBy = '';
+
+      if (state.expandIcon) {
+        const hasExpandLabel = state.expandIcon['aria-label'] || state.expandIcon['aria-labelledby'];
+        // If there is no explicit aria-label, calculate default accName attribute for expandIcon button,
+        // using the following steps:
+        // 1. If there is an aria-label, it is "Open [aria-label]"
+        // 2. If there is an aria-labelledby, it is "Open [aria-labelledby target]" (using aria-labelledby + ids)
+        // 3. If there is no aria-label/ledby attr, it falls back to "Open"
+        // We can't fall back to a label/htmlFor name because of https://github.com/w3c/accname/issues/179
+        const defaultOpenString = 'Open'; // this is english-only since it is the fallback
```

Comment:
Can you add a TODO comment here? I'm not sure it would be ideal to have an english fallback... ideally we'd have this provided by client somehow, for now is fine I guess.

Reply:
This is the same logic Sarah used for computing the combobox aria-label/aria-labelledby, and it's only used in the worst case where the client does not provide anything :)

Reply:
I'd say ideally we would have to start enforcing the client to provide something, but since that would be a breaking change we won't do that. Therefore, let's definitely do the way you're doing, it just would be nice to have a TODO comment to remind ourselves about this problem on a next major version, when we could maybe start enforcing it

Reply:
Yeah. we do not have a closed solution for this for sure (what worries me the most is the label element, otherwise `aria-labelledby` or `aria-label` is quite easy to enforce by type)., perhaps we could have a context to indicate labelling by an element.

Anyway, going default to english sounds like a huge assumption for a design system. We're giving a default for clients and they don't have to worry, but reality is that they should worry and we should warn them about not providing a proper label.

I don't think there's a solution at this moment, that's why I don't mind this, but we should definitely label this with something to indicate that investigations would be nice, just my two cent, this is a nit.

Reply:
yeah, it definitely feels like a damned if you do, damned if you don't situation 😅